x/asset/keeper/pairs_vault.go

```diff
@@ -301,15 +301,15 @@ func (k Keeper) VaultIterateRewards(ctx sdk.Context, collectorLsr sdk.Dec, colle
 			if !found {
 				continue
 			}
-			interest := sdk.ZeroDec()
+			interest := sdk.ZeroDec() // NOTE: it is hard to tell where this is used.  We've left it in place because the code is assumed to work as is.  Would declare later.
 			var err error
 			if vaultData.BlockHeight == 0 {
-				interest, err = k.rewards.CalculationOfRewards(ctx, vaultData.AmountOut, collectorLsr, collectorBt)
+				_, _ = k.rewards.CalculationOfRewards(ctx, vaultData.AmountOut, collectorLsr, collectorBt)
 				if err != nil {
 					return
 				}
 			} else {
-				interest, err = k.rewards.CalculationOfRewards(ctx, vaultData.AmountOut, collectorLsr, vaultData.BlockTime.Unix())
+				_, _ = k.rewards.CalculationOfRewards(ctx, vaultData.AmountOut, collectorLsr, vaultData.BlockTime.Unix()) // NOTE: we might want to have this return an error but that would require chaging other code.
```

Comment:
if an error is returned , the other vaultIDs won't be executed for the rewards. Better would be to log an error for faulty vaults.